x-pack/plugin/write-load-forecaster/src/main/java/org/elasticsearch/xpack/writeloadforecaster/LicensedWriteLoadForecaster.java

```diff
@@ -0,0 +1,145 @@
+import org.elasticsearch.core.TimeValue;
+import org.elasticsearch.index.Index;
+import org.elasticsearch.index.shard.IndexWriteLoad;
+
+import java.util.OptionalDouble;
+import java.util.OptionalLong;
+import java.util.function.BooleanSupplier;
+
+class LicensedWriteLoadForecaster implements WriteLoadForecaster {
+    public static final Setting<TimeValue> MAX_INDEX_AGE_SETTING = Setting.timeSetting(
+        "write_load_forecaster.max_index_age",
+        TimeValue.timeValueDays(7),
+        TimeValue.timeValueHours(1),
+        Setting.Property.NodeScope,
+        Setting.Property.Dynamic
+    );
+
+    public static final Setting<TimeValue> MIN_UPTIME_SETTING = Setting.timeSetting(
+        "write_load_forecaster.min_uptime",
+        TimeValue.timeValueHours(1),
+        TimeValue.timeValueMinutes(1),
+        Setting.Property.NodeScope,
+        Setting.Property.Dynamic
+    );
+
+    private final BooleanSupplier hasValidLicense;
+    private volatile TimeValue maxIndexAge;
+    private volatile TimeValue minShardUptime;
+
+    LicensedWriteLoadForecaster(BooleanSupplier hasValidLicense, Settings settings, ClusterSettings clusterSettings) {
+        this(hasValidLicense, MAX_INDEX_AGE_SETTING.get(settings), MIN_UPTIME_SETTING.get(settings));
+        clusterSettings.addSettingsUpdateConsumer(MAX_INDEX_AGE_SETTING, this::setMaxIndexAgeSetting);
+        clusterSettings.addSettingsUpdateConsumer(MIN_UPTIME_SETTING, this::setMinUptimeSetting);
+    }
+
+    LicensedWriteLoadForecaster(BooleanSupplier hasValidLicense, TimeValue maxIndexAge, TimeValue minShardUptime) {
+        this.hasValidLicense = hasValidLicense;
+        this.maxIndexAge = maxIndexAge;
+        this.minShardUptime = minShardUptime;
+    }
+
+    private void setMaxIndexAgeSetting(TimeValue updatedMaxIndexAge) {
+        this.maxIndexAge = updatedMaxIndexAge;
+    }
+
+    private void setMinUptimeSetting(TimeValue updatedMinShardUptime) {
+        this.minShardUptime = updatedMinShardUptime;
+    }
+
+    @Override
+    public ClusterState withWriteLoadForecastForWriteIndex(String dataStreamName, ClusterState clusterState) {
+        if (hasValidLicense.getAsBoolean() == false) {
+            return clusterState;
+        }
+
+        DataStream dataStream = clusterState.metadata().dataStreams().get(dataStreamName);
+
+        if (dataStream == null) {
+            return clusterState;
+        }
+
+        double maxWriteLoadAvg = Double.MIN_VALUE;
+        double totalWriteLoadAvg = 0;
+        double totalIndices = 0;
+
+        for (Index index : dataStream.getIndices()) {
+            final IndexMetadata indexMetadata = clusterState.metadata().getIndexSafe(index);
+            final long indexAge = System.currentTimeMillis() - indexMetadata.getCreationDate();
+            final IndexWriteLoad writeLoad = indexMetadata.getWriteLoad();
+
+            if (index.equals(dataStream.getWriteIndex()) || indexAge > maxIndexAge.millis() || writeLoad == null) {
```

Comment:
For rollovers that happen less frequently than `maxIndexAge`, I think we'd want to include the index. I think we also want to include the first index in creation order that has creation-date before the max-age (though that may be unnecessary if we just ensure to include minimum 1). The reason being that in order to look back max-age time, you need to include the last index that was created before max-age since the life time of that overlaps the lookback period.